app/src/main/java/fr/free/nrw/commons/quiz/QuizChecker.java

```diff
@@ -0,0 +1,144 @@
+package fr.free.nrw.commons.quiz;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.support.v7.app.AlertDialog.Builder;
+
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.WelcomeActivity;
+import fr.free.nrw.commons.mwapi.MediaWikiApi;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+/**
+ * to check whether to pop quiz or not
```

Comment:
It would be good to have more detailed Javadocs, especially for classes - the reader should be able to have a fairly detailed idea of what the class does based on the Javadocs alone. For instance, see this example: http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#examples (I personally think that is a bit much, we certainly don't need 20+ line docs, lol. But more detail than the present is needed).

How about something like "Fetches the number of images the user has uploaded and the number of images reverted, in order to calculate the % of reverts. Then calls quiz if the revert parameter exceeds the required percentage."